src/services/timeline.rs

```diff
@@ -0,0 +1,78 @@
+use crate::models::{
+    goal::{Day, TimeFilter},
+    slot::Slot,
+    timeline::Timeline,
+};
+
+impl Timeline {
+    pub fn apply_filter(&mut self, _filter: &Option<TimeFilter>) -> Timeline {
+        todo!("apply_filter not implemented");
+    }
+
+    /// Filtering timeline based on before_time and after_time fields in TimeFilter
+    pub fn filter_timing(&mut self, _before_time: Option<usize>, _after_time: Option<usize>) {
+        todo!("filter_timing not implemented");
+    }
+
+    /// Filtering timeline based on on_days field in TimeFilter
+    pub fn filter_on_days(&mut self, _days_to_filter: &[Day]) {
+        todo!("filter_on_days not implemented");
+    }
+
+    /// Filtering timeline based on not_on field in TimeFilter
+    pub fn filter_not_on(&mut self, _slots_to_filter: &[Slot]) {
+        todo!("filter_not_on not implemented");
+    }
+}
+
+/// Validate that a given value is valid time number which must be between 0 and 24
+fn _validate_time(time: Option<usize>, time_name: &str) {
```

Comment:
Good to add this check!  
Isn't a more appropriate place to do this when the input gets deserialized?

Reply:
That will be better, to refuse in appropriete/wrong input from users. But I think this will be developed when adding error handling features like Issue https://github.com/tijlleenders/ZinZen-scheduler/issues/299